### PR TITLE
Tour: "Singleton Objects"

### PR DESCRIPTION
**Pull request by S11001001**
With links from other tour parts.

This is incompatible with #307, but can be renumbered with its script straightforwardly.

### CODE REVIEW

File: tutorials/tour/singleton-objects.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Singleton objects are sort of a shorthand for defining a single-use class, which can't directly be instantiated, and a `val` member at the point of definition of the `object`, with the same name. Indeed, like `val`s, singleton objects can be defined as members of a [trait](traits.html) or class, though this is atypical.
+
+A singleton object can extend classes and traits. In fact, a [case class](case-classes.html) with no [type parameters](generic-classes.html) will by default create a singleton object of the same name, with a [`Function*`](http://www.scala-lang.org/api/current/scala/Function1.html) trait implemented.
+
+## Companions ##
+
+Most singleton objects do not stand alone, but instead are associated with a class of the same name. The “singleton object of the same name” of a case class, mentioned above, is an example of this. When this happens, the singleton object is called the *companion object* of the class, and the class is called the *companion class* of the object. This is so common that it's safe to refer to singleton objects as “companion objects” informally, without much risk of confusion.
```
Comment (SethTisue) [inline tutorials/tour/singleton-objects.md]: I don't agree with this last sentence; it seems incorrect and misleading to me to call something a "companion object" unless it actually is one. Let's not encourage this.

**Comment by SethTisue** [comment]
LGTM except for the one piece of feedback. @S11001001 if you agree and correct it, I'll merge.

**Comment by heathermiller** [comment]
Thanks for the review @SethTisue! I'm going to pull this in and add a follow-up commit where I remove the last contentious sentence. That way this is in the docs without any further delay :)

File: tutorials/tour/singleton-objects.md (hunk 1 above)
Comment (S11001001) [inline tutorials/tour/singleton-objects.md]: @SethTisue I agree.

**Comment by S11001001** [comment]
@SethTisue @heathermiller Thanks for reviewing and merging!

**Pull request merged**